### PR TITLE
Digeq 54 routing based on jump to section

### PR DESCRIPTION
**Pull request by warrenbailey**
**What**
Initial routing based on jump conditions. This pull request will allow the use to jump sections depending on the answers to a question.

**How to test**
1. Checkout out this branch
2. Run the server
3. Browse to the debug questionnaire: http://127.0.0.1:5000/questionnaire/1/ADSEDD?debug=True
4. Answer Episode 1 to question 2
5. Notice that you are routed to section 3 rather than section 2

**Who can test**
Anyone apart from @warren-methods and @weapdiv-david

### CODE REVIEW

File: branching.py [hunk 1]
```diff
@@ -0,0 +1,13 @@
+class Condition(object):
+    def __init__(self, *args):
+        self.target = args[0]
+        self.trigger = args[1]
+        self.state = args[2]
+
```
Comment (LJBabbage) [inline branching.py:6]: Can't see any use for the function below

File: questionnaireManagerTest.py [hunk 2]
```diff
@@ -268,38 +274,40 @@ def test_progress(self):
...
-
-        assert q_manager.get_total_questions() == 2
-
-        q2 = q_manager.get_next_question()
-
-        q_manager.jump_to_question('start')
-
-        current_question = q_manager.get_current_question()
-        assert current_question.reference == q1.reference
-        assert q_manager.get_current_question_index() == 1
```
Comment (LJBabbage) [inline questionnaireManagerTest.py:302]: Please remove redundant code

File: srunner.py [hunk 3]
```diff
@@ -78,66 +78,74 @@ def questionnaire_viewer(questionnaire_id, quest_session_id=None):
...
 
-        return render_template('questions/' + question.type + '.html',
-                                question=question,
-                                user_response=user_responses,
-                                questionnaire=q_manager,
-                                request=request)
     else:
-        return render_template('survey_intro.html',
-                                questionnaire=q_manager,
-                                request=request)
```
Comment (LJBabbage) [inline srunner.py:140]: Please remove

File: questionnaireManagerTest.py (hunk 2 above)
Comment (warrenbailey) [inline questionnaireManagerTest.py:302]: Done

File: branching.py (hunk 1 above)
Comment (warrenbailey) [inline branching.py:6]: Done

File: srunner.py (hunk 3 above)
Comment (warrenbailey) [inline srunner.py:140]: Done

**Pull request merged**